Cargo.toml

```diff
@@ -30,6 +30,7 @@ approx = "0.5"
 itertools = "0.13.0"
 ahash = "0.8.11"
 rayon = "1.10"
+uuid = { version = "1.16", features = ["v4"] }
```

Comment:
Should we add the `fast-rng` feature too? Iirc it depends on rand which we're already using so there shouldn't be a huge compile time impact